src/sentry/static/sentry/app/views/events/utils/eventsRequest.tsx

```diff
@@ -112,45 +157,38 @@ class EventsRequest extends React.PureComponent {
     end: null,
     interval: '1d',
     limit: 15,
-    getCategory: i => i,
     query: '',
 
     includePrevious: true,
     includeTransformedData: true,
   };
 
-  constructor(props) {
-    super(props);
-    this.state = {
-      reloading: false || props.loading,
-      timeseriesData: null,
-    };
-  }
+  state = {
+    reloading: false || !!this.props.loading,
+    timeseriesData: null,
+  };
 
   componentDidMount() {
     this.fetchData();
   }
-
-  componentDidUpdate(prevProps) {
+  componentDidUpdate(prevProps: EventsRequestProps) {
     if (isEqual(omitIgnoredProps(prevProps), omitIgnoredProps(this.props))) {
       return;
     }
-
     this.fetchData();
   }
-
   componentWillUnmount() {
     this.unmounting = true;
   }
 
+  unmounting: boolean = false;
```

Comment:
Could we mark this as private? (assuming we don't want to access it from outside)